src/TraceEvent/Symbols/SymbolReader.cs

```diff
@@ -2681,16 +2681,16 @@ unsafe internal SourceFile(SymbolModule module, IDiaSourceFile sourceFile)
             // 1 CALG_MD5 checksum generated with the MD5 hashing algorithm.
             // 2 CALG_SHA1 checksum generated with the SHA1 hashing algorithm.
             m_hashType = sourceFile.checksumType;
-            if (m_hashType != 2 && m_hashType != 1 && m_hashType != 0)
-            {
-                Debug.Assert(false, "Unknown hash type");
-                m_hashType = 0;
-            }
             if (m_hashType != 0)
             {
+                uint hashSizeInBytes;
+                fixed (byte* bufferPtr = m_hash)
+                    sourceFile.get_checksum(0, out hashSizeInBytes, out *bufferPtr);
+
                 // MD5 is 16 bytes
                 // SHA1 is 20 bytes  
-                m_hash = m_hashType == 1 ? new byte[16] : new byte[20];
+                // SHA-256 is 32 bytes
+                m_hash = new byte[hashSizeInBytes];
```

Comment:
The API has the concept of having a hash checksum or not.   (see HashCheckSum, and does nothing if it does not have a checksum).   We want this code to fall back to this 'don't do anything' if it sees a hash that it does not understand (in fact if you had not run debug, the existing code would have given you the source (just not given you reassurance that it was the right source).

Currently HasCheckSum is defined to be m_hash != null, which means by setting this unconditionally here, we are saying we always have a checksum, which is not true.

My recommendation:

Create a method 'GetCryptoProvider() that returns the Crypto Provider.
Make ComputeHash take that as a parameter.  
Make HasChecksum be GetCryptoProivder() != null.